src/uct/tcp/sockcm/sockcm_iface.c

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2017-2019.  ALL RIGHTS RESERVED.
+ * Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "sockcm_iface.h"
+#include "sockcm_ep.h"
+#include <uct/base/uct_worker.h>
+#include <ucs/sys/string.h>
+#include <ucs/sys/sock.h>
+
+enum uct_sockcm_process_event_flags {
+    UCT_SOCKCM_PROCESS_EVENT_DESTROY_SOCK_ID_FLAG = UCS_BIT(0),
+    UCT_SOCKCM_PROCESS_EVENT_ACK_EVENT_FLAG       = UCS_BIT(1)
+};
+
+static ucs_config_field_t uct_sockcm_iface_config_table[] = {
+    {"BACKLOG", "1024",
+     "Maximum number of pending connections for a (listening?) socket.",
+     ucs_offsetof(uct_sockcm_iface_config_t, backlog), UCS_CONFIG_TYPE_UINT},
+
+    {"SOCK_ID_QUOTA", "64",
+     "How many sockcm connections can progress simultaneously.",
+     ucs_offsetof(uct_sockcm_iface_config_t, sock_id_quota), UCS_CONFIG_TYPE_UINT},
+
+    {NULL}
+};
+
+static UCS_CLASS_DECLARE_DELETE_FUNC(uct_sockcm_iface_t, uct_iface_t);
+
+static ucs_status_t uct_sockcm_iface_query(uct_iface_h tl_iface,
+                                           uct_iface_attr_t *iface_attr)
+{
+    memset(iface_attr, 0, sizeof(uct_iface_attr_t));
+
+    iface_attr->iface_addr_len  = sizeof(ucs_sock_addr_t);
+    iface_attr->device_addr_len = 0;
+    iface_attr->cap.flags       = UCT_IFACE_FLAG_CONNECT_TO_SOCKADDR    |
+                                  UCT_IFACE_FLAG_CB_ASYNC               |
+                                  UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE;
+    /* User's private data size is UCT_SOCKCM_UDP_PRIV_DATA_LEN minus room for
+     * the private_data header (to hold the length of the data) */
+    iface_attr->max_conn_priv   = UCT_SOCKCM_MAX_CONN_PRIV;
+
+    return UCS_OK;
+}
+
+static int uct_sockcm_iface_is_reachable(const uct_iface_h tl_iface,
+                                         const uct_device_addr_t *dev_addr,
+                                         const uct_iface_addr_t *iface_addr)
+{
+    /* Reachability can be checked with the uct_md_is_sockaddr_accessible API call */
+    return 1;
+}
+
+static ucs_status_t uct_sockcm_iface_get_address(uct_iface_h tl_iface, uct_iface_addr_t *iface_addr)
+{
+    ucs_sock_addr_t *sockcm_addr = (ucs_sock_addr_t *)iface_addr;
+
+    sockcm_addr->addr    = NULL;
+    sockcm_addr->addrlen = 0;
+    return UCS_OK;
+}
+
+static ucs_status_t uct_sockcm_iface_accept(uct_iface_h tl_iface,
+                                            uct_conn_request_h conn_request)
+{
+    return UCS_OK;
```

Comment:
maybe return `UCS_ERR_NOT_IMPLEMENTED` instead of `UCS_OK` for all skeleton funcs?